pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -285,76 +305,92 @@ func (l *ListOptionIndexer) ListByOptions(ctx context.Context, lo ListOptions, p
 	}
 
 	if len(orderByClauses) > 0 {
-		stmt += "\n  ORDER BY "
-		stmt += strings.Join(orderByClauses, ", ")
+		query += "\n  ORDER BY "
+		query += strings.Join(orderByClauses, ", ")
 	} else {
 		// make sure one default order is always picked
 		if l.namespaced {
-			stmt += "\n  ORDER BY f.\"metadata.namespace\" ASC, f.\"metadata.name\" ASC "
+			query += "\n  ORDER BY f.\"metadata.namespace\" ASC, f.\"metadata.name\" ASC "
 		} else {
-			stmt += "\n  ORDER BY f.\"metadata.name\" ASC "
+			query += "\n  ORDER BY f.\"metadata.name\" ASC "
 		}
 	}
 
-	// 3- Pagination: LIMIT clause (from lo.Pagination and/or lo.ChunkSize/lo.Resume)
+	// 4- Pagination: LIMIT clause (from lo.Pagination and/or lo.ChunkSize/lo.Resume)
+
+	// before proceeding, save a copy of the query and params without LIMIT/OFFSET
+	// for COUNTing all results later
+	countQuery := fmt.Sprintf("SELECT COUNT(*) FROM (%s)", query)
```

Comment:
I saw this as well, and it looked like `query` is safe, but I'd prefer using a safer statement-prep form as a matter of style

Reply:
@MbolotSuse not from my understanding, as query is (supposed to be) independently safe.

@ericpromislow I do not understand your proposal. `query` needs to be string-built, as opposed to prepared, because table and column names are variable - and so does `countQuery` by consequence.

How would you propose to use a prepared statement in this case?